docs/docs/k8s/ingress.md

```diff
@@ -0,0 +1,363 @@
+---
+
+# Kubernetes Ingress Controller
+
+Use Pomerium as a first-class secure-by-default Ingress Controller. Dynamically provision routes from Ingress resources and set policy based on annotations. By defining routes as Ingress resources you can independently create and remove them from Pomerium's configuration. This enables workflows more native to Kubernetes environments such as actions based on pull requests.
+
+## Prerequisites
+
+- A certificate management solution. If you do not already have one in place, this article covers using Cert Manager.
+- A Redis backend with high-persistence is highly recommended.
+
+### System Requirements
+
+- Kubernetes v1.19.0+
+- Pomerium [Helm Chart](https://github.com/pomerium/pomerium-helm/tree/master/charts/pomerium) v25.0.0+
+
+### Limitations
+
+::: warning
+
+Only one Ingress Controller instance/replica is supported per Pomerium cluster.
+
+:::
+
+## Installation
+
+### Helm
+
+Our instructions for [Installing Pomerium Using Helm](/docs/k8s/helm.md) includes the Ingress Controller as part of the documented configuration.
+
+
+```yaml
+ingressController:
+  enabled: true
+```
+
+### Docker Image
+
+You may deploy your own manifestations by using the `pomerium/ingress-controller` docker image.
+
+## Configuration
+
+|  Flag                          | Description                                                          |
+| ------------------------------ | -------------------------------------------------------------------- |
+| `--databroker-service-url`     | the databroker service url
+| `--databroker-tls-ca`          | base64 encoded tls CA
+| `--databroker-tls-ca-file`     | tls CA file path for the databroker connection connection
+| `--health-probe-bind-address`  | The address the probe endpoint binds to. (default ":8081")
+| `--metrics-bind-address`       | The address the metric endpoint binds to. (default ":8080")
+| `--name`                       | IngressClass controller name (default "pomerium.io/ingress-controller")
+| `--namespaces`                 | namespaces to watch, or none to watch all namespaces
+| `--prefix`                     | Ingress annotation prefix (default "ingress.pomerium.io")
+| `--shared-secret`              | base64-encoded shared secret for communicating with databroker
+| `--update-status-from-service` | update ingress status from given service status (pomerium-proxy)|
+
+The helm chart exposes a subset of these flags for appropriate customization.
+
+## Usage
+
+### Defining Routes
+
+If you've tested Pomerium using the all-in-one service, you're probably familiar with configuring routes in Pomerium's `config.yaml`. In this environment, each route is defined as an Ingress resource in the Kubernetes API.
+The Ingress Controller will monitor Ingress resources in the cluster, creating a Pomerium route definition for each one.  Policy and other configuration options for the route are set by using annotations starting with `ingress.pomerium.io/`.
```

Comment:
Can we link to the thing we are referencing here (the all in one service guide)?

Reply:
> familiar with configuring routes in Pomerium's `config.yaml`.
if not, can we link them to the docs that outline the routes ad policies reference docs?

Reply:
> In this environment

`this` being an Ingress Controller?

Reply:
Updated to specify the subject.